### PR TITLE
enable linspace & logspace tests

### PR DESCRIPTION
**Pull request by barney-s**
Enable linspace tests and fix the test failures

1. fix _copy to handle self shape=(1) and copy src shape=0, ref: https://github.com/pytorch/xla/issues/7505#issuecomment-2400509352
2. for squeeze where we expect the self's shape to change, dont use copy_ instead replace.
3. in test_ops reset dtype to float when an int64 is passed for linspace
   case. This is to workaround known pytorch failure: https://github.com/pytorch/pytorch/issues/137546

ref:
* https://github.com/pytorch/xla/issues/7505#issuecomment-2400509352
* https://github.com/pytorch/xla/issues/7505#issuecomment-2400895692
* https://github.com/pytorch/xla/issues/7505#issuecomment-2403039282
* https://github.com/pytorch/pytorch/issues/137546

fixes: https://github.com/pytorch/xla/issues/7505, https://github.com/pytorch/xla/issues/7507

### CODE REVIEW

File: experimental/torch_xla2/test/test_ops.py [hunk 1]
```diff
@@ -213,6 +212,7 @@ def run_export_and_compare(testcase,
         sample_input.input, sample_input.args, sample_input.kwargs))
       with testcase.env:
         res2 = func(input2, *args2, **kwargs2)
+        print("input2: ", input2, "sample_input", sample_input.input, " res2:  ", res2)
```
Comment (ManfeiBai) [inline experimental/torch_xla2/test/test_ops.py:215]: thanks, do we want to keep these `print` code here?

File: experimental/torch_xla2/test/test_ops.py [hunk 2]
```diff
@@ -224,6 +224,7 @@ def run_export_and_compare(testcase,
               rtol=rtol,
               equal_nan=equal_nan, check_output=check_output)
         else:
+          print(func.name, "res: ", res, "res2: ", res2)
```
Comment (ManfeiBai) [inline experimental/torch_xla2/test/test_ops.py:227]: thanks, do we want to keep these `print` code here?

File: experimental/torch_xla2/test/test_ops.py [hunk 3]
```diff
@@ -269,6 +270,7 @@ def test_reference_eager(self, device, dtype, op):
         continue
       check_output = op.name not in random_ops
 
+      print("[DEBUG] sample_input: ", sample_input)
```
Comment (ManfeiBai) [inline experimental/torch_xla2/test/test_ops.py:273]: thanks, do we want to keep these `print` code here?

File: experimental/torch_xla2/test/test_ops.py (hunk 2 above)
Comment (barney-s) [reply experimental/torch_xla2/test/test_ops.py:227]: No. Still WIP.

File: experimental/torch_xla2/torch_xla2/ops/jaten.py [hunk 4]
```diff
@@ -37,7 +37,8 @@
   torch.ops.aten.uniform_: torch.ops.aten.uniform,
   torch.ops.aten.relu_: torch.ops.aten.relu,
   torch.ops.aten.normal_: torch.ops.aten.normal,
-  torch.ops.aten.squeeze_: torch.ops.aten.squeeze,
+  # squeeze_ is expected to change tensor's shape. So replace with new value 
+  torch.ops.aten.squeeze_: (torch.ops.aten.squeeze, True),
```
Comment (barney-s) [inline experimental/torch_xla2/torch_xla2/ops/jaten.py:41]: Error without this change: 
```
E         
E         To execute this test, run the following from the base repo dir:
E              python test/test_ops.py -k TestOpInfoCPU.test_reference_eager_quantile_cpu_float32
E         
E         This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0

test/test_ops.py:182: AssertionError
------------------------------------------------------------------ Captured stdout call -------------------------------------------------------------------
[DEBUG] sample_input:  SampleInput(input=-7.407406330108643, args=(0.5,), kwargs={}, broadcasts_input=False, name='')
input2:  XLATensor2(<class 'jaxlib.xla_extension.ArrayImpl'> -7.4074063) sample_input tensor(-7.4074)  res2:   XLATensor2(<class 'jaxlib.xla_extension.ArrayImpl'> [-7.4074063])
quantile res:  tensor(-7.4074) res2:  tensor([-7.4074])
```

File: experimental/torch_xla2/test/test_ops.py (hunk 3 above)
Comment (barney-s) [reply experimental/torch_xla2/test/test_ops.py:273]: commenting this line out. But retaining the commented code for future debugging. This is a commonly used debug line.

File: experimental/torch_xla2/test/test_ops.py (hunk 1 above)
Comment (barney-s) [reply experimental/torch_xla2/test/test_ops.py:215]: removed.

File: experimental/torch_xla2/test/test_ops.py (hunk 2 above)
Comment (barney-s) [reply experimental/torch_xla2/test/test_ops.py:227]: removed.

**Review by qihqi — APPROVED** [review]
Please rebase before merging, thanks!

**Review by ManfeiBai — APPROVED** [review]
Thanks, LGTM

**Pull request merged**